projects/04_cube_xr/CubeXr.cpp

```diff
@@ -330,36 +342,31 @@ void CubeXrApp::Render()
     submitInfo.commandBufferCount = 1;
     submitInfo.ppCommandBuffers   = &frame.cmd;
     // No need to use semaphore when XR is enabled.
-    if (IsXrEnabled()) {
-        submitInfo.waitSemaphoreCount   = 0;
-        submitInfo.ppWaitSemaphores     = nullptr;
-        submitInfo.signalSemaphoreCount = 0;
-        submitInfo.ppSignalSemaphores   = nullptr;
-    }
-    else {
+    if (!IsXrEnabled()) {
         submitInfo.waitSemaphoreCount   = 1;
         submitInfo.ppWaitSemaphores     = &frame.imageAcquiredSemaphore;
         submitInfo.signalSemaphoreCount = 1;
         submitInfo.ppSignalSemaphores   = &frame.renderCompleteSemaphore;
     }
     submitInfo.pFence = frame.renderCompleteFence;
-
+    swapchain->Wait(imageIndex);
     PPX_CHECKED_CALL(GetGraphicsQueue()->Submit(&submitInfo));
 
-    // No need to present when XR is enabled.
     if (!IsXrEnabled()) {
         PPX_CHECKED_CALL(swapchain->Present(imageIndex, 1, &frame.renderCompleteSemaphore));
     }
     else {
```

Comment:
not sure if it is a good idea, but just want to hear your opinions :): since the `Present` is not a real Present in the case of XR swapchain, and the all the args are ignored anyways, maybe it is cleaner to have just 1 unified `Present` code path here for both XR and non-XR? Because otherwise, maybe it is clearer to have another XR specific function here instead of naming it `Present` with ignored inputs?